src/cbmc/cbmc_parse_options.cpp

```diff
@@ -662,6 +662,8 @@ int cbmc_parse_optionst::get_goto_program(
     if(!binaries.empty())
       config.set_from_symbol_table(symbol_table);
 
+    config.set_addr_bits(cmdline, symbol_table);
```

Comment:
Setting the config has already been a multistage process before, in particular there are side conditions that the symbol table has been populated. Ie. You don't get the final config before having processed all input files. I see two options: 1) update the config after loading input files (means hooking into language_Ui.final and read Goto binary), 2) introduce a config.finalise call that must be invoked after having read the last input file.
The first option wouldn't require any collaboration of *_parse_optionst; the second one does require collaboration but makes it a bit more uniform wrt how it is currently done in master.